### PR TITLE
Add strictness option to a JenkinsQueueJob

### PR DESCRIPTION
**Pull request by kuleshovilya**
**Task name**: JenkinsQueueJobV2

**Description**: Add input boolean for determining whether to consider *unstable* result as part of a success or not

**Documentation changes required:** Y

**Added unit tests:** N

**Attached related issue:** [15031](https://github.com/microsoft/azure-pipelines-tasks/issues/15031)

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/JenkinsQueueJobV2/task.json [hunk 1]
```diff
@@ -101,6 +101,15 @@
...
+        },
+        {
+            "name": "strictDefinition",
+            "type": "boolean",
+            "label": "Strict success definition",
+            "defaultValue": false,
+            "required": false,
+            "helpMarkDown": "Specifies strictness of a success definition: whether to consider unstable as a failure or not. False for non-strict, and true for strict version."
+
         }
```
Comment (DaniilShmelev) [inline Tasks/JenkinsQueueJobV2/task.json:113]: Maybe we should move this input to the "advanced" group?

```suggestion
        {
            "name": "strictDefinition",
            "type": "boolean",
            "label": "Strict success definition",
            "defaultValue": false,
            "required": false,
            "helpMarkDown": "Specifies strictness of a success definition: whether to consider unstable as a failure or not. False for non-strict, and true for strict version.",
            "groupName": "advanced"
        }
```

**Review by DaniilShmelev — COMMENTED** [review]
Please also include task-loc.json changes :)

File: Tasks/JenkinsQueueJobV2/task.json (hunk 1 above)
Comment (kuleshovilya) [reply Tasks/JenkinsQueueJobV2/task.json]: Yeah, seems like a logical thing to do

File: Tasks/JenkinsQueueJobV2/job.ts [hunk 2]
```diff
@@ -218,7 +218,7 @@ export class Job {
             return tl.TaskResult.Succeeded;
         } else if (this.State === JobState.Done) {
             const resultCode = this.ParsedExecutionResult.result.toUpperCase();
-            if (resultCode == 'SUCCESS' || resultCode == 'UNSTABLE') {
+            if (resultCode == 'SUCCESS' || (resultCode == 'UNSTABLE' && this.queue.TaskOptions.strictDefinition)) {
```
Comment (tatyana-kostromskaya) [inline Tasks/JenkinsQueueJobV2/job.ts:221]: Should not it be here `!this.queue.TaskOptions.strictDefinition`?

Comment (kuleshovilya) [reply Tasks/JenkinsQueueJobV2/job.ts:221]: > Please also include task-loc.json changes :)

Done

**Comment by kuleshovilya** [comment]
> Please also include task-loc.json changes :)

Done

File: Tasks/JenkinsQueueJobV2/job.ts (hunk 2 above)
Comment (kuleshovilya) [reply Tasks/JenkinsQueueJobV2/job.ts:221]: Yes, my bad, supposed to be NOT

File: Tasks/JenkinsQueueJobV2/task.loc.json [hunk 3]
```diff
@@ -101,6 +101,15 @@
         "resizable": "true",
         "rows": "4"
       }
+    },
+    {
+      "name": "strictDefinition",
```
Comment (anatolybolshakov) [inline Tasks/JenkinsQueueJobV2/task.loc.json:106]: Could it make sense to make it more specific, like 'failOnUnstableResultStatus'?

File: Tasks/JenkinsQueueJobV2/task.json [hunk 4]
```diff
@@ -101,6 +101,16 @@
...
+        },
+        {
+            "name": "strictDefinition",
+            "type": "boolean",
+            "label": "Strict success definition",
+            "defaultValue": false,
+            "required": false,
+            "helpMarkDown": "Specifies strictness of a success definition: whether to consider unstable as a failure or not. False for non-strict, and true for strict version.",
+            "groupName": "advanced"
+
```
Comment (DaniilShmelev) [inline Tasks/JenkinsQueueJobV2/task.json:113]: NIT: please remove this empty line

File: Tasks/JenkinsQueueJobV2/task.loc.json (hunk 3 above)
Comment (DaniilShmelev) [reply Tasks/JenkinsQueueJobV2/task.loc.json:106]: @kuleshovilya Probably should also update the label if you decide to change this

**Review by DaniilShmelev — APPROVED** [review]
Take a look at the comments, other than that LGTM!

File: Tasks/JenkinsQueueJobV2/task.loc.json (hunk 3 above)
Comment (kuleshovilya) [reply Tasks/JenkinsQueueJobV2/task.loc.json:106]: Yeah, something like that would sound better, although I think we can shorten it a bit to 'failOnUnstableResult', status is implied anyways.

Comment (anatolybolshakov) [reply Tasks/JenkinsQueueJobV2/task.loc.json:106]: @kuleshovilya yes, make sense 👍

**Review by tatyana-kostromskaya — APPROVED** [review]

**Pull request merged**